env.go

```diff
@@ -138,6 +137,9 @@ type runnable interface {
 	// Start tells Runnable to start.
 	Start() error
 
+	// RunOneOff tells Runnable to start as batch job and wait until completion with output capture.
+	RunOneOff(ctx context.Context) (output string, err error)
```

Comment:
nit: I find the naming confusing here. Suggestions `RunOnce`, `RunOnceOff` or borrowing kube terminology `RunAsJob`

Reply:
I like those, what do you prefer?